dangerfile.js

```diff
@@ -0,0 +1,26 @@
+// eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars
+import { danger, warn, fail, message } from 'danger';
+
+const modifiedChangelog = danger.git.modified_files.filter((filePath) => {
+  const srcFilePattern = /^packages\/([a-z-])*\/CHANGELOG.md/i;
+  return srcFilePattern.test(filePath);
+});
+
+const modifiedSrcFiles = danger.git.modified_files.filter((filePath) => {
+  const srcFilePattern = /^packages\/([a-z-])*\/src/i;
+  return srcFilePattern.test(filePath);
+});
+
+const hasCHANGELOGChanges = modifiedChangelog.length > 0;
+const hasModifiedSrcFiles = modifiedSrcFiles.length > 0;
+
+// Fail if there are src code changes without a CHANGELOG
+if (hasModifiedSrcFiles && !hasCHANGELOGChanges) {
```

Comment:
Out of curiosity, does danger.git.modified_files return the number of file changes? I wonder if there would ever be a case where we would have both `hasModifiedSrcFiles` & `hasCHANGLELOGChanges` but it didn't capture the full scope of the changes.